lib/plurimath/math/function/ternary_function.rb

```diff
@@ -112,6 +134,80 @@ def latex_wrap(field)
             latex
           end
         end
+
+        def narypr(function_symbol, function_type: "undOvr")
+          chr_arg     = { "m:val": function_symbol }
+          limloc_arg  = { "m:val": function_type }
+          subhide_arg = { "m:val": "0" }
+          suphide_arg = { "m:val": "0" }
+          chr_tag     = Utility.ox_element("chr", attributes: chr_arg, namespace: "m")
+          limloc_tag  = Utility.ox_element("limLoc", attributes: limloc_arg, namespace: "m")
+          subhide_tag = Utility.ox_element("subHide", attributes: subhide_arg, namespace: "m")
+          suphide_tag = Utility.ox_element("supHide", attributes: suphide_arg, namespace: "m")
+          nary_pr_tag = Utility.ox_element("naryPr", namespace: "m")
+          Utility.update_nodes(
+            nary_pr_tag,
+            [
+              chr_tag,
+              limloc_tag,
+              subhide_tag,
+              suphide_tag,
+            ],
+          )
+        end
+
+        def sub_parameter
+          sub_tag = Utility.ox_element("sub", namespace: "m")
+          return empty_tag(sub_tag) unless parameter_one
+
+          Utility.update_nodes(sub_tag, insert_t_tag(parameter_one))
+        end
+
+        def sup_parameter
+          sup_tag = Utility.ox_element("sup", namespace: "m")
+          return empty_tag(sup_tag) unless parameter_two
+
+          Utility.update_nodes(sup_tag, insert_t_tag(parameter_two))
+        end
+
+        def e_parameter
+          e_tag = Utility.ox_element("e", namespace: "m")
+          return empty_tag(e_tag) unless parameter_three
+
+          Utility.update_nodes(e_tag, insert_t_tag(parameter_three))
+        end
+
+        def empty_tag(wrapper_tag)
+          r_tag = Utility.ox_element("r", namespace: "m")
+          r_tag << (Utility.ox_element("t", namespace: "m") << "&#8203;")
+          wrapper_tag << r_tag
+        end
+
+        def insert_t_tag(parameter)
+          parameter_value = parameter.to_omml_without_math_tag
+          r_tag = Utility.ox_element("r", namespace: "m")
+          if parameter.is_a?(Symbol)
+            r_tag << (Utility.ox_element("t", namespace: "m") << parameter_value)
+            [r_tag]
+          elsif parameter.is_a?(Number)
+            Utility.update_nodes(r_tag, parameter_value)
+            [r_tag]
+          else
+            Array(parameter_value)
+          end
+        end
+
+        def underover_class?(field)
+          if field.is_a?(Symbol)
+            ["&#x22c0;", "&#x22c1;", "&#x22c2;", "&#x22c3;"].include?(field.value)
+          else
+            ["ubrace", "obrace"].include?(field.class_name)
+          end
```

Comment:
I think wherever we are using `if .is_a?(Classname)` type check can be moved to methods in the respective class. 
That way here we can simply do something like `field.underover_class?` here.